components/survey_monkey/survey_monkey.app.mjs

```diff
@@ -73,25 +103,86 @@ export default {
 
       return axios($ || this, config);
     },
-    async getSurveys() {
-      return await this._makeRequest({
+    async _paginatedRequest(params) {
+      // https://api.surveymonkey.net/v3/docs?javascript#pagination
+      // using default SurveyMonkey pagination - 50 per page
+      const amountPerPage = 50;
+      const values = [];
+
+      const { path } = params;
+
+      let page = 1;
+      let response;
+
+      do {
+        response = await this._makeRequest({
+          ...params,
+          path: `${path}?page=${page}&per_page=${amountPerPage}`,
+        });
+
+        if (response.data) values.push(...response.data);
+        else throw new Error(response);
+      } while (page++ < Math.ceil(response.total / amountPerPage));
+
+      return values;
+    },
```

Comment:
To simplify a bit more taking advantage of `axios` properties:
```suggestion
    async _paginatedRequest(args = {}) {
      // https://api.surveymonkey.net/v3/docs?javascript#pagination
      // using default SurveyMonkey pagination - 50 per page
      const amountPerPage = 50;
      const values = [];

      let page = 1;
      let response;

      do {
        response = await this._makeRequest({
          params: {
            page,
            per_page: amountPerPage,
          },
          ...args,
        });

        if (response.data) {
          values.push(...response.data);
        } else {
          throw new Error(response);
        }
      } while (page++ < Math.ceil(response.total / amountPerPage));

      return values;
    },
```

Reply:
Great suggestion! I overlooked the use of axios `params` completely. Changed to use them as per your suggestion.